patches/minecraft/net/minecraft/village/Village.java.patch

```diff
@@ -0,0 +1,72 @@
+--- ../src-base/minecraft/net/minecraft/village/Village.java
++++ ../src-work/minecraft/net/minecraft/village/Village.java
+@@ -26,7 +26,7 @@
+ import net.minecraft.util.math.Vec3d;
+ import net.minecraft.world.World;
+ 
+-public class Village
++public class Village implements net.minecraftforge.common.capabilities.ICapabilitySerializable<NBTTagCompound>
+ {
+     private World field_75586_a;
+     private final List<VillageDoorInfo> field_75584_b = Lists.<VillageDoorInfo>newArrayList();
+@@ -43,11 +43,13 @@
+ 
+     public Village()
+     {
++        this(null);
+     }
+ 
+     public Village(World p_i1675_1_)
+     {
+         this.field_75586_a = p_i1675_1_;
++        this.capabilities = net.minecraftforge.event.ForgeEventFactory.gatherCapabilities(this);
+     }
+ 
+     public void func_82691_a(World p_82691_1_)
+@@ -466,6 +468,8 @@
+                 this.field_82693_j.put(nbttagcompound1.func_74779_i("Name"), Integer.valueOf(nbttagcompound1.func_74762_e("S")));
+             }
+         }
++        if (this.capabilities != null && p_82690_1_.func_74764_b("ForgeCaps")) this.capabilities.deserializeNBT(p_82690_1_.func_74775_l("ForgeCaps"));
++
+     }
+ 
+     public void func_82689_b(NBTTagCompound p_82689_1_)
+@@ -522,6 +526,7 @@
+         }
+ 
+         p_82689_1_.func_74782_a("Players", nbttaglist1);
++        if (this.capabilities != null) p_82689_1_.func_74782_a("ForgeCaps", this.capabilities.serializeNBT());
+     }
+ 
+     public void func_82692_h()
+@@ -553,4 +558,29 @@
+             this.field_75590_b = p_i1674_3_;
+         }
+     }
++
```

Comment:
Could use a 
`/* ======================================== FORGE START =====================================*/`
section for good measure